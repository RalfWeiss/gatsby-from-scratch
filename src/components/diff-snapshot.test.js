const React = require("react");
const renderer = require("react-test-renderer");
// const styled = require('styled-components')
import styled from "styled-components";
const { toMatchDiffSnapshot } = require("snapshot-diff");

const Component = ({ test, className }) => <p {...{ className }}>{test}</p>;

const StyledComponent = styled(Component)`
  background-color: blue;
`;

expect.extend({ toMatchDiffSnapshot });

test("snapshot difference between 2 React components state", () => {
  expect(<Component test="say" />).toMatchDiffSnapshot(
    <Component test="my name" />
  );
});

test("snapshot component", () => {
  expect(<Component test="say" />).toMatchSnapshot();
});

test("inlineSnapshot styled component", () => {
  expect(<StyledComponent test="say" />).toMatchInlineSnapshot(`
    <ForwardRef(Styled(Component))
      test="say"
    />
  `);
});

test("snapshot styled component", () => {
  expect(<StyledComponent test="say" />).toMatchSnapshot();
});


it("match expected snapsho", () => {
  const propValue = "my value";
  const tree = renderer.create(<Component test={propValue} />).toJSON();
  const expectedSnapshot = `
    <p>
      ${propValue}
    </p>
`;
  // expect(tree).toMatchSnapshot();
  expect(tree).toMatchInlineSnapshot(expectedSnapshot);
});
