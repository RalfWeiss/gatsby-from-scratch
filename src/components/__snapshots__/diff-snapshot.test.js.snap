// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot component 1`] = `
<Component
  test="say"
/>
`;

exports[`snapshot difference between 2 React components state 1`] = `
"Snapshot Diff:
- <Component test=\\"say\\" />
+ <Component test=\\"my name\\" />

  <p>
-   say
+   my name
  </p>"
`;

exports[`snapshot styled component 1`] = `
<ForwardRef(Styled(Component))
  test="say"
/>
`;
